packages/chart/src/ChartUtils.ts

```diff
@@ -884,142 +408,640 @@ class ChartUtils {
+
+  addSourcesToSeriesData(
+    seriesDataParam: Partial<PlotData>,
+    plotStyle: SeriesPlotStyle,
+    sources: SeriesDataSource[],
+    axisTypeMap: AxisTypeMap
+  ): void {
+    const seriesData = seriesDataParam;
+    for (let k = 0; k < sources.length; k += 1) {
+      const source = sources[k];
+      const { axis, type: sourceType } = source;
+
+      const dataAttributeName = this.getPlotlyProperty(plotStyle, sourceType);
+      set(seriesData, dataAttributeName, []);
+
+      const axisProperty =
+        axis != null ? this.getAxisPropertyName(axis.type) : null;
+      if (axisProperty != null) {
+        const axes = axisTypeMap.get(axis.type);
+        if (axes) {
+          const axisIndex = axes.indexOf(axis);
+          const axisIndexString = axisIndex > 0 ? `${axisIndex + 1}` : '';
+          seriesData[
+            `${axisProperty}axis`
+          ] = `${axisProperty}${axisIndexString}`;
+        }
+      }
+    }
+  }
+
+  addStylingToSeriesData(
+    seriesDataParam: Partial<PlotData>,
+    plotStyle: SeriesPlotStyle,
+    theme: typeof ChartTheme = ChartTheme,
+    lineColor: string | null = null,
+    shapeColor: string | null = null,
+    shape: string | null = null,
+    shapeSize: number | null = null,
+    seriesVisibility: 'legendonly' | boolean | null = null
+  ): void {
+    const { dh } = this;
+    const seriesData = seriesDataParam;
+    // Add some empty objects so we can fill them in later with details without checking for existence
+    seriesData.marker = { line: {} }; // border line width on markers
+    seriesData.line = {
+      width: 1, // default line width for lines, should eventually be able to override
+    };
+
+    if (plotStyle === dh.plot.SeriesPlotStyle.AREA) {
+      seriesData.fill = 'tozeroy';
+    } else if (plotStyle === dh.plot.SeriesPlotStyle.STACKED_AREA) {
+      seriesData.stackgroup = 'stack';
+    } else if (plotStyle === dh.plot.SeriesPlotStyle.STEP) {
+      seriesData.line.shape = 'hv'; // plot.ly horizontal then vertical step styling
+    } else if (plotStyle === dh.plot.SeriesPlotStyle.HISTOGRAM) {
+      // The default histfunc in plotly is 'count', but the data passed up from the API provides explicit x/y values and bins
+      // Since it's converted to bar, just set the widths of each bar
+      seriesData.width = [];
+
+      if (seriesData.marker.line !== undefined) {
+        Object.assign(seriesData.marker.line, {
+          color: theme.paper_bgcolor,
+          width: 1,
+        });
+      }
+    } else if (plotStyle === dh.plot.SeriesPlotStyle.OHLC) {
+      (seriesData as Partial<OhlcData>).increasing = {
+        line: { color: theme.ohlc_increasing },
+      };
+      (seriesData as Partial<OhlcData>).decreasing = {
+        line: { color: theme.ohlc_decreasing },
+      };
+    } else if (plotStyle === dh.plot.SeriesPlotStyle.PIE) {
+      seriesData.textinfo = 'label+percent';
+
+      // TODO Open DefinitelyTyped/Plotly PR to mark family and size as optional
```

Comment:
Looks like we never opened a PR for this...

Reply:
I don't have enough information about this, who should have opened a PR?

Reply:
It would have been from when Tony was doing TS conversion. If you could create a ticket for it just with this comment and the link to the pie.d.ts, that would be great.